axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveServiceImpl.java

```diff
@@ -1429,4 +1432,57 @@ public void updateStockMoveMobility(StockMove stockMove, BigDecimal movedQty, Un
     }
     stockMoveRepo.save(stockMove);
   }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void updateStockMoveDestinationLocation(StockMove stockMove, StockLocation toStockLocation)
+      throws AxelorException {
+    if (stockMove.getStatusSelect() != StockMoveRepository.STATUS_CANCELED
+        && stockMove.getStatusSelect() != StockMoveRepository.STATUS_REALIZED) {
+      if (toStockLocation != null) {
+        stockMove.setToStockLocation(toStockLocation);
+        stockMove.setToAddress(toStockLocation.getAddress());
+        Beans.get(StockMoveToolService.class).computeAddressStr(stockMove);
+        stockMoveRepo.save(stockMove);
+      } else {
+        throw new AxelorException(
+            stockMove,
+            TraceBackRepository.CATEGORY_MISSING_FIELD,
+            I18n.get(IExceptionMessage.STOCK_MOVE_6));
+      }
+    } else {
+      throw new AxelorException();
+    }
+  }
+
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void addLineStockMove(
+      StockMove stockMove,
+      Product product,
+      TrackingNumber trackingNumber,
+      BigDecimal qty,
+      BigDecimal realQty,
+      Unit unit,
+      Integer conformitySelect)
+      throws AxelorException {
+
+    StockMoveLine newLine = new StockMoveLine();
+    newLine.setStockMove(stockMove);
+    newLine.setLineTypeSelect(StockMoveLineRepository.TYPE_NORMAL);
+    newLine.setProduct(product);
+    stockMoveLineService.setProductInfo(stockMove, newLine, stockMove.getCompany());
+
+    if (product.getTrackingNumberConfiguration() != null && trackingNumber != null) {
+      newLine.setTrackingNumber(trackingNumber);
+    }
+    newLine.setQty(qty);
+    newLine.setRealQty(realQty);
+    newLine.setUnit(unit);
+    newLine.setConformitySelect(conformitySelect);
+
+    stockMoveLineService.compute(newLine, stockMove);
+
+    stockMoveLineRepo.save(newLine);
+  }
```

Comment:
This method is only creating and saving a new stock move. This method must be moved in StockMoveLineService and called "createStockMoveLine", and maybe calls existing `createStockMoveLine` methods.